certbot/compat/os.py

```diff
@@ -26,7 +26,8 @@
     if not hasattr(ourselves, attribute):
         setattr(ourselves, attribute, getattr(std_os, attribute))
 
-# Similar to os.path, allow certbot.compat.os.path to behave as a module
+# Import our path module, then allow certbot.compat.os.path to behave as a module (like os.path)
+from certbot.compat import path  # type: ignore  # pylint: disable=wrong-import-position
```

Comment:
If we add this `type: ignore`, is all type checking disabled on all calls to all functions in `certbot.compat.os.path`? I suspect this is the case which doesn't seem great. Is there an easy way to avoid this `type: ignore`?

Reply:
https://mypy.readthedocs.io/en/latest/common_issues.html#spurious-errors-and-locally-silencing-the-checker, see in particular the note in this section.

I think mypy will continue to typecheck members of path even if ignore is set on this line.

Reply:
Let's give a demo of this behavior.

```python
# util.py
from os import *  # type: ignore
def chmod():
    pass
```

```python
# main.py
from util import chmod
from util import mkdir
from util import nonexistent
```

```bash
$ mypy main.py
main.py:3: error: Module 'util' has no attribute 'nonexistent'
```

Despite `# type: ignore` is set on the star import from `os` in `util`, mypy successfully sees that `mkdir` exists in `util` thanks to the star import, and that `nonexistent` does not exist in `util`.

Reply:
Thanks for the explanation! Typechecking indeed still seems to work fine so I think we can leave things as is.